akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala

```diff
@@ -13,83 +16,88 @@ import MediaTypes._
 
-    "(without headers)" in test { accept ⇒
-      accept(`text/plain`) should select(`text/plain`, `UTF-8`)
+    "(without headers)" test { accept ⇒
+      //accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: */*" in test { accept ⇒
+    "Accept: */*" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: */*;q=.8" in test { accept ⇒
+    "Accept: */*;q=.8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: text/*" in test { accept ⇒
+    "Accept: text/*" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/xml` withCharset `UTF-16`) should select(`text/xml`, `UTF-16`)
       accept(`audio/ogg`) should reject
     }
 
-    "Accept: text/*;q=.8" in test { accept ⇒
+    "Accept: text/*;q=.8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/xml` withCharset `UTF-16`) should select(`text/xml`, `UTF-16`)
       accept(`audio/ogg`) should reject
     }
 
-    "Accept: text/*;q=0" in test { accept ⇒
+    "Accept: text/*;q=0" test { accept ⇒
       accept(`text/plain`) should reject
       accept(`text/xml` withCharset `UTF-16`) should reject
       accept(`audio/ogg`) should reject
     }
 
-    "Accept-Charset: UTF-16" in test { accept ⇒
+    "Accept-Charset: UTF-16" test { accept ⇒
+      accept(`text/plain`) should select(`text/plain`, `UTF-16`)
+      accept(`text/plain` withCharset `UTF-8`) should reject
+    }
+
+    "manually created Accept-Charset: UTF-16" in testHeaders(headers.`Accept-Charset`(Vector(HttpCharsets.`UTF-16`.toRange))) { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-16`)
       accept(`text/plain` withCharset `UTF-8`) should reject
     }
 
-    "Accept-Charset: UTF-16, UTF-8" in test { accept ⇒
+    "Accept-Charset: UTF-16, UTF-8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept-Charset: UTF-8;q=.2, UTF-16" in test { accept ⇒
+    "Accept-Charset: UTF-8;q=.2, UTF-16" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-16`)
       accept(`text/plain` withCharset `UTF-8`) should select(`text/plain`, `UTF-8`)
     }
 
-    "Accept-Charset: UTF-8;q=.2" in test { accept ⇒
-      accept(`text/plain`) should select(`text/plain`, `ISO-8859-1`)
+    "Accept-Charset: UTF-8;q=.2" test { accept ⇒
+      // FIXME: which rule would enforce that behavior?
+      // accept(`text/plain`) should select(`text/plain`, `ISO-8859-1`)
```

Comment:
LGTM! I was playing around with that one as well and arrived at the same result. ;)

Looks like the logic changed in that we now reject instead of falling back to the default charset (`ISO-8859-1` back then) in that case.